clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -447,10 +449,12 @@ else if (!future.isRetriable())
         return true;
     }
 
+    // reset join-group future in order to retry the join-group request
     private synchronized void resetJoinGroupFuture() {
         this.joinFuture = null;
     }
 
+    // reset state to UNJOINED and request re-join
```

Comment:
Do these comments have much value? They basically just describe the two lines of code below.

Reply:
This is commented by @abbccdda who are trying to clarify the different functionalities here, after second thought I think it makes more sense to just comment on the original call for different cases.